src/libraries/System.Reflection.Emit.ILGeneration/ref/System.Reflection.Emit.ILGeneration.cs

```diff
@@ -65,12 +65,13 @@ public abstract class ILGenerator
         public static bool operator ==(System.Reflection.Emit.Label a, System.Reflection.Emit.Label b) { throw null; }
         public static bool operator !=(System.Reflection.Emit.Label a, System.Reflection.Emit.Label b) { throw null; }
     }
-    public sealed partial class LocalBuilder : System.Reflection.LocalVariableInfo
+    public abstract class LocalBuilder : System.Reflection.LocalVariableInfo
     {
-        internal LocalBuilder() { }
+        protected LocalBuilder() { }
         public override bool IsPinned { get { throw null; } }
         public override int LocalIndex { get { throw null; } }
         public override System.Type LocalType { get { throw null; } }
+        public abstract MethodInfo Method { get; }
```

Comment:
@jkotas @AaronRobinsonMSFT @buyaa-n What if the local is in a constructor?  I would expect `MethodBase` here, not `MethodInfo`

Reply:
That is a good point. I was also interested in why this sort of property wasn't on the base class, `LocalVariableInfo`, but didn't raise it since API review had already occurred. I think the constructor case though does seem to warrant further consideration.

@buyaa-n or @steveharter Was the constructor case discussed in the API review? I didn't attend so perhaps I am missing context.

Reply:
I think we should drop the property from public surface and keep it as an internal detail in the managed RefEmit.

Reply:
> @buyaa-n or @steveharter Was the constructor case discussed in the API review? 

No, existing runtime implementation creates a MethodBuilder for constructors, i.e. they are kept as a method that is why it works, not sure how mono handle constructors

Reply:
I see. The issue here then becomes `ConstructorBuilder.GetILGenerator()` and the path to this API is now broken because one could not get `ConstructorInfo` when starting from a `ConstructorBuilder`.

> I think we should drop the property from public surface and keep it as an internal detail in the managed RefEmit.

I'm starting to agree.

Reply:
The method builder created for constructors is internal implementation detail. I do not think that this method builder is meant to leak out through the public surface. It should always show up wrapper by `ConstructorBuilder`.

Reply:
@jkotas I agree with your statement. In order to be concrete, are you using that statement as a way to further demonstrate that `public abstract MethodInfo Method { get; }` should not be part of the public API and instead an internal implementation detail?

Reply:
Yep. The current API shape is not right and the easiest way to fix that is to avoid exposing the API in the public surface.

Reply:
Added a comment for [removing the property](https://github.com/dotnet/runtime/issues/93497#issuecomment-1781514695) in the approved API, will remove it form this PR